sherpa/astro/models/__init__.py

```diff
@@ -1025,6 +1025,7 @@ def __init__(self, name='beta2d'):
         ArithmeticModel.__init__(self, name,
                                  (self.r0, self.xpos, self.ypos, self.ellip,
                                   self.theta, self.ampl, self.alpha))
+        self.cache = 0
```

Comment:
So, this could be applied in `sherpa.models.model.RegriddableModel2D` - e.g. as a class attribute, in the same way that `ndim` is set - which would mean we don't need to apply it to each 2D model individually (at least for those that derive from it, which I expect to be all of them).

I think this would be cleaner, but it wouldn't surprise me if it had some issue somewhere else.

Reply:
The problem is that `cache` is a property in a class that is a parent of `RegriddableModel2D`. If I simply add the line `cache=0` to `RegriddableModel2D`, then I'm overwriting that property. In other classes, we added `self.cache=5` to the `__init__` in the Baseclass. Then, it would set that through the property. However, in the 2D models, the base class `__init__` is never called thus that approach won't work unless I add a `super.__init__(...)` for every `__init__` function. It would probably be cleaner to add that call, but it's also more invasive than I wanted to be in this PR.

Reply:
Ta for checking. It doesn't help that our classes are often caught between the two approaches you can use in Python - either be explicit about the parent class name or use `super()` - which confuses things a bit (there is a reason to be explicit, i.e. `super` is not always the answer) and this is definitely not the time to go down that bandwagon!